### PR TITLE
ADMU CI Migration

### PR DESCRIPTION
**Pull request by jworkmanjc**
## Issues
* [SA-3609](https://jumpcloud.atlassian.net/browse/SA-3609) - ADMU CI Migration

## What does this solve?

This is a non-impactful patch release to update the build/test/release workflow in GitHub Actions. A number of items in the repository directory have been cleaned up, I've removed files that were saved in git history but not necessary to be stored in the repository.

Files Removed:

- circleCi/config.yml — This was the point of the PR to remove the CircleCI workflows
- Deploy/ADMU.ps1 — This file is generated before we create the .exe files, it's not necessary to track it in git (but open to discussion of this)
- jumpcloud-ADMU/Jumpcloud-ADMU.nuspec — This file is generated in build and we'd never need to upload it
- Pester Test Results — These are unnecessary to store in git history

Modified CI Steps:

The build steps have generally been consolidated into a single `build.ps1` file. It's easy to run locally, and the idea is that you'd run `build.ps1` once before you commit changes to see if there's anything additional worth committing to git history (generally this would be moduleChangeLog.md + the module psd1). Regardless this runs on CI **and** in the Release CI workflow. If there was a better way to save artifacts between **workflows** we could cut that part out of Release CI and only run this once in PR CI. This would be preferred and it is possible but I want to break that into a different card.

Changes to Nuspec/Nupkg:

I've cleaned up the file structure in the nuspec/ nupkg files. If you run build.ps1 you'll see the nuspec file generated should only contain docs, help files, Start-Migration and the other required module information. We used to package up a number of unnecessary items in the module which was not best practice. There are other items we should add like a LICENSE, readme file and bundle those into our package. I want to break that out into a different card.

Changes to Tests:

Tests are broken up by several different runners, GHA runners are slower than CircleCI by default so running certain tests in parallel seems to cut down on the total runtime. I'd like to further break out these tests in a future card.

Build Tests should now throw if you haven't run `build.ps1` on a given day before committing changes. I want our changelog + psd1 generation date to be the same as whatever date we release on. It just looks nicer so it's included. Check these tests out and let me know if you want to change anything. 

## Is there anything particularly tricky?

After review we'll change two items to the Release CI to only run that workflow on merge to main branch. We have experience doing this with the PWSH module so that step should work when we next merge. 

## How should this be tested?

View the two most recent actions logs, the CI and Release CI workflows are both triggered with the last commit. 

## Screenshots

[SA-3609]: https://jumpcloud.atlassian.net/browse/SA-3609?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# ADMU Code Owners
+
+/PowerShell/ @TheJumpCloud/jc-sa
```
Comment (kmaranionjc) [inline .github/CODEOWNERS:3]: Giving unknown owner error

File: ModuleChangelog.md [hunk 2]
```diff
@@ -1,3 +1,17 @@
+## 2.5.1
+
+Release Date: October 25, 2023
```
Comment (kmaranionjc) [inline ModuleChangelog.md:3]: If released today, change date

File: jumpcloud-ADMU/Powershell/Tests/SetLastLoggedOnUserTest.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    write-host "Importing Start-Migration Script:"
+    . $PSScriptRoot\..\Start-Migration.ps1
+    # setup tests (This creates any of the users in the build vars dictionary)
+    write-host "Running SetupAgent Script:"
+    . $PSScriptRoot\SetupAgent.ps1
+    # End region for test user generation
+
+    # Remove users with ADMU_ prefix
+    # Remove Created Users
+    #Get-JCuser -username "ADMU_*" | Remove-JCuser -Force
```
Comment (kmaranionjc) [inline jumpcloud-ADMU/Powershell/Tests/SetLastLoggedOnUserTest.Tests.ps1:29]: Should we include this?

Comment (kmaranionjc) [inline jumpcloud-ADMU/Powershell/Tests/SetLastLoggedOnUserTest.Tests.ps1:29]: Should we include this?

**Review by kmaranionjc — COMMENTED** [review]
Dropped couple of comments. Biggest thing is should we do delete jc user on the new test files

**Review by kmaranionjc — DISMISSED** [review]

**Review by kmaranionjc — APPROVED** [review]

**Review by gweinjc — APPROVED** [review]

**Pull request merged**